File: python/2.7/factory/VeggiePizza.py
from Pizza import *

class VeggiePizza(Pizza):
    def __init__(self, ingredientFactory):
        super(VeggiePizza, self).__init__()
        self.ingredientFactory = ingredientFactory

    def prepare(self):
        print "Preparing %s" %(self.name)
        self.dough = self.ingredientFactory.createDough()
        self.sauce = self.ingredientFactory.createSauce()
        self.cheese = self.ingredientFactory.createCheese()
        self.veggies = self.ingredientFactory.createVeggies()

File: js/node/factory/com/pizza/pizza/VeggiePizza.js
require('./Pizza');

VeggiePizza = function(ingredientFactory) {
    Pizza.apply(this);
    this.ingredientFactory = ingredientFactory;
}
VeggiePizza.prototype = new Pizza();
VeggiePizza.prototype.prepare = function() {
    console.log("Prepareing " + this.name);
    this.dough = this.ingredientFactory.createDough();
    this.sauce = this.ingredientFactory.createSauce();
    this.cheese = this.ingredientFactory.createCheese();
    this.veggies = this.ingredientFactory.createVeggies();
}